src/PointwiseFunctions/AnalyticSolutions/Xcts/ConstantDensityStar.hpp

```diff
@@ -0,0 +1,189 @@
+ * \f]
+ * and real parameters \f$\alpha\f$ and \f$\beta\f$ that are determined by
+ * the following relations:
+ * \f[
+ * \rho_0 R^2=\frac{3}{2\pi}f^2(\alpha) \quad \text{with}
+ * \quad f(\alpha)=\frac{\alpha^5}{(1+\alpha^2)^3} \\
+ * \frac{\beta}{R} + 1 = C u_\alpha(R)
+ * \f]
+ *
+ * This solution is described in detail in \cite Baumgarte2006ug since it
+ * exhibits the non-uniqueness properties that are typical for the XCTS system.
+ * In the simple case of the constant-density star the non-uniqueness is
+ * apparent from the function \f$f(\alpha)\f$, which has two solutions for any
+ * \f$\rho_0\f$ smaller than a critical density
+ * \f[
+ * \rho_\mathrm{crit}=\frac{3}{2\pi R^2}\frac{5^2}{6^6}
+ * \approx\frac{0.0320}{R^2} \text{,}
+ * \f]
+ * a unique solution for \f$\rho_0=\rho_\mathrm{crit}\f$ and no solutions
+ * above the critical density \cite Baumgarte2006ug . The authors identify the
+ * \f$\alpha < \alpha_\mathrm{crit}=\sqrt{5}\f$ and \f$\alpha >
+ * \alpha_\mathrm{crit}\f$ branches of solutions with the strong-field and
+ * weak-field regimes, respectively (see \cite Baumgarte2006ug for details).
+ * In this implementation we compute the weak-field solution by choosing the
+ * \f$\alpha > \alpha_\mathrm{crit}\f$ that corresponds to the density
+ * \f$\rho_0\f$ of the star. Therefore, we supply initial data
+ * \f$\psi_\mathrm{init}=1\f$ so that a nonlinear iterative numerical solver
+ * will converge to the same weak-field solution.
+ */
+class ConstantDensityStar {
+ private:
+  struct Density {
+    using type = double;
+    static constexpr OptionString help{"The constant density within the star"};
+    static double lower_bound() noexcept { return 0.; }
+  };
+  struct Radius {
+    using type = double;
+    static constexpr OptionString help{"The conformal radius of the star"};
+    static double lower_bound() noexcept { return 0.; }
+  };
+
+ public:
+  using options = tmpl::list<Density, Radius>;
+  static constexpr OptionString help{
+      "A constant density star in general relativity"};
+
+  ConstantDensityStar() = default;
+  ConstantDensityStar(const ConstantDensityStar&) noexcept = delete;
+  ConstantDensityStar& operator=(const ConstantDensityStar&) noexcept = delete;
+  ConstantDensityStar(ConstantDensityStar&&) noexcept = default;
+  ConstantDensityStar& operator=(ConstantDensityStar&&) noexcept = default;
+  ~ConstantDensityStar() noexcept = default;
+
+  ConstantDensityStar(double density, double radius,
+                      const OptionContext& context = {});
+
+  double density() const noexcept { return density_; }
+  double radius() const noexcept { return radius_; }
+
+  // @{
+  /// Retrieve variable at coordinates `x`
+  auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<Xcts::Tags::ConformalFactor<DataVector>> /*meta*/) const
+      noexcept -> tuples::TaggedTuple<Xcts::Tags::ConformalFactor<DataVector>>;
+
+  static auto variables(
```

Comment:
They return constants, so don't depend on any parameters. They can be called on a class instance the same as the non-static functions.

Reply:
of course, but that is not a reason to mark them static; I can not think of any use case for calling these functions without a class instance...

Reply:
right, made them non-static now.

ping @moxcodes just to make you aware of this, since you suggested this change for the Poisson/Lorentzian solution